Description: Correctly check build architecture.
Author: Michael Meskes <meskes@debian.org>

--- a/configure
+++ b/configure
@@ -381,7 +381,7 @@ test_execute_path()
 check_environment()
 {
   test_header environment
-  BUILD_CPU=`uname -m`
+  BUILD_CPU=`dpkg-architecture -qDEB_HOST_GNU_CPU`
   [ "$OS" = "solaris" ] && BUILD_CPU=`isainfo | cut -f 1 -d ' '`
   case "$BUILD_CPU" in
     i[3456789]86|x86|i86pc|BePC)
