Description: Make virtualbox-ose work with kernel > 2.6.36
Author: Oracle
--- virtualbox-ose/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c.r66523	2010-10-28 20:30:38.556181086 +0400
+++ virtualbox-ose/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c	2010-10-28 20:33:08.726181061 +0400
@@ -1337,6 +1337,54 @@
     return fRc;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 36)
+/**
+ * Helper for detecting TAP devices.
+ */
+static bool vboxNetFltIsTapDevice(PVBOXNETFLTINS pThis, struct net_device *pDev)
+{
+    if (pDev->ethtool_ops && pDev->ethtool_ops->get_drvinfo)
+    {
+        struct ethtool_drvinfo Info;
+
+        memset(&Info, 0, sizeof(Info));
+        Info.cmd = ETHTOOL_GDRVINFO;
+        pDev->ethtool_ops->get_drvinfo(pDev, &Info);
+        Log3(("vboxNetFltIsTapDevice: driver=%s version=%s bus_info=%s ",
+              Info.driver, Info.version, Info.bus_info));
+
+        return !strncmp(Info.driver,   "tun", 4)
+            && !strncmp(Info.bus_info, "tap", 4);
+    }
+
+    return false;
+}
+
+/**
+ * Helper for updating the link state of TAP devices.
+ * Only TAP devices are affected.
+ */
+static void vboxNetFltSetTapLinkState(PVBOXNETFLTINS pThis, struct net_device *pDev, bool fLinkUp)
+{
+    if (vboxNetFltIsTapDevice(pThis, pDev))
+    {
+        Log3(("vboxNetFltSetTapLinkState: bringing %s tap device link state ",
+              fLinkUp ? "up" : "down"));
+        netif_tx_lock_bh(pDev);
+        if (fLinkUp)
+            netif_carrier_on(pDev);
+        else
+            netif_carrier_off(pDev);
+        netif_tx_unlock_bh(pDev);
+    }
+}
+#else /* LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 36) */
+DECLINLINE(void) vboxNetFltSetTapLinkState(PVBOXNETFLTINS pThis, struct net_device *pDev, bool fLinkUp)
+{
+    /* Nothing to do for pre-2.6.36 kernels. */
+}
+#endif /* LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 36) */
+
 /**
  * Internal worker for vboxNetFltLinuxNotifierCallback.
  *
@@ -1378,6 +1426,12 @@
 #endif
 
     /*
+     * If attaching to TAP interface we need to bring the link state up
+     * starting from 2.6.36 kernel.
+     */
+    vboxNetFltSetTapLinkState(pThis, pDev, true);
+
+    /*
      * Set indicators that require the spinlock. Be abit paranoid about racing
      * the device notification handle.
      */
@@ -1741,6 +1795,8 @@
 
     if (fRegistered)
     {
+        vboxNetFltSetTapLinkState(pThis, pDev, false);
+
 #ifndef VBOXNETFLT_LINUX_NO_XMIT_QUEUE
         skb_queue_purge(&pThis->u.s.XmitQueue);
 #endif
--- virtualbox-ose/src/VBox/Additions/linux/drm/vboxvideo_drm.c	2010-10-13 11:43:00.000000000 +0200
+++ virtualbox-ose/src/VBox/Additions/linux/drm/vboxvideo_drm.c	2010-11-22 10:59:31.000000000 +0100
@@ -78,8 +78,11 @@
 	/* .driver_features = DRIVER_USE_MTRR, */
 	.load = vboxvideo_driver_load,
 	.reclaim_buffers = drm_core_reclaim_buffers,
+	/* As of Linux 2.65.37, always the internal functions are used. */ 
+#if LINUX_VERSION_CODE < KERNEL_VERSION (2, 6, 37) 
 	.get_map_ofs = drm_core_get_map_ofs,
 	.get_reg_ofs = drm_core_get_reg_ofs,
+#endif
 	.fops = {
 		 .owner = THIS_MODULE,
 		 .open = drm_open,
