Description: Make Xsession.d script ignore errors.
Author: Michael Meskes <meskes@debian.org>

Index: virtualbox/src/VBox/Additions/x11/Installer/98vboxadd-xclient
===================================================================
--- virtualbox.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient	2013-11-19 03:40:59.137475714 -0500
+++ virtualbox/src/VBox/Additions/x11/Installer/98vboxadd-xclient	2013-11-19 03:40:59.133475714 -0500
@@ -42,11 +42,11 @@
     no_display=1
 fi
 
-/usr/bin/VBoxClient --clipboard
-/usr/bin/VBoxClient --checkhostversion
+/usr/bin/VBoxClient --clipboard || true
+/usr/bin/VBoxClient --checkhostversion || true
 test -z "$no_display" &&
-    /usr/bin/VBoxClient --display
+    /usr/bin/VBoxClient --display || true
 test -z "$no_display" &&
-    /usr/bin/VBoxClient --seamless
+    /usr/bin/VBoxClient --seamless || true
 test -z "$no_display" &&
-    /usr/bin/VBoxClient --draganddrop
+    /usr/bin/VBoxClient --draganddrop || true
