Description: Made error message tell about missing vboxnetflt module.
Author: Michael Meskes <meskes@debian.org>

Index: virtualbox/src/VBox/Devices/Network/DrvIntNet.cpp
===================================================================
--- virtualbox.orig/src/VBox/Devices/Network/DrvIntNet.cpp
+++ virtualbox/src/VBox/Devices/Network/DrvIntNet.cpp
@@ -1987,7 +1987,7 @@
             return VERR_PDM_NO_ATTACHED_DRIVER;
         }
         return PDMDrvHlpVMSetError(pDrvIns, rc, RT_SRC_POS,
-                                   N_("Failed to open/create the internal network '%s'"), pThis->szNetwork);
+                                   N_("Failed to open/create the internal network '%s' (you might need to modprobe vboxnetflt to make it accessible)"), pThis->szNetwork);
     }
 
     AssertRelease(OpenReq.hIf != INTNET_HANDLE_INVALID);
