Description: Fix ARCH setting in modules Makefiles.
Author: Michael Meskes <meskes@debian.org>, Gianfranco Costamagna <locutusofborg@debian.org>

--- a/src/VBox/Installer/linux/Makefile-header.gmk
+++ b/src/VBox/Installer/linux/Makefile-header.gmk
@@ -60,7 +60,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell uname -m)),)
+ifeq ($(filter-out x86_64 amd64 AMD64,$(shell dpkg-architecture -qDEB_HOST_GNU_CPU)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
