Description: Fix ARCH setting in modules Makefiles.
Author: Michael Meskes <meskes@debian.org>

Index: virtualbox/src/VBox/HostDrivers/Support/linux/Makefile
===================================================================
--- virtualbox.orig/src/VBox/HostDrivers/Support/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
+++ virtualbox/src/VBox/HostDrivers/Support/linux/Makefile	2013-11-19 03:41:05.817476025 -0500
@@ -44,9 +44,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
Index: virtualbox/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile
===================================================================
--- virtualbox.orig/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
+++ virtualbox/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile	2013-11-19 03:41:05.817476025 -0500
@@ -35,9 +35,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
Index: virtualbox/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile
===================================================================
--- virtualbox.orig/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
+++ virtualbox/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
@@ -35,9 +35,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
Index: virtualbox/src/VBox/HostDrivers/VBoxPci/linux/Makefile
===================================================================
--- virtualbox.orig/src/VBox/HostDrivers/VBoxPci/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
+++ virtualbox/src/VBox/HostDrivers/VBoxPci/linux/Makefile	2013-11-19 03:41:05.821476025 -0500
@@ -35,9 +35,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
Index: virtualbox/src/VBox/Installer/linux/Makefile.include.header
===================================================================
--- virtualbox.orig/src/VBox/Installer/linux/Makefile.include.header	2013-11-19 03:41:05.821476025 -0500
+++ virtualbox/src/VBox/Installer/linux/Makefile.include.header	2013-11-19 03:41:05.821476025 -0500
@@ -54,9 +54,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+    ifeq ($(ARCH),x86)
+      ifeq ($(CONFIG_X86_32),y)
+        BUILD_TARGET_ARCH := x86
+      else
+        BUILD_TARGET_ARCH := amd64
+      endif
+    else
+      BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+    endif
   endif
  endif
 else
