Description: Fix ARCH setting in modules Makefiles.
Author: Michael Meskes <meskes@debian.org>

diff -urNad virtualbox-ose-3.0.8-dfsg~/src/VBox/Additions/linux/sharedfolders/Makefile.module virtualbox-ose-3.0.8-dfsg/src/VBox/Additions/linux/sharedfolders/Makefile.module
--- virtualbox-ose-3.0.8-dfsg~/src/VBox/Additions/linux/sharedfolders/Makefile.module	2009-10-07 11:27:13.000000000 +0200
+++ virtualbox-ose-3.0.8-dfsg/src/VBox/Additions/linux/sharedfolders/Makefile.module	2009-10-07 13:43:30.705472033 +0200
@@ -33,9 +33,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+    ifeq ($(ARCH),x86)
+      ifeq ($(CONFIG_X86_32),y)
+        BUILD_TARGET_ARCH := x86
+      else
+        BUILD_TARGET_ARCH := amd64
+      endif
+    else
+      BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+    endif
   endif
  endif
 else
diff -urNad virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/Support/linux/Makefile virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/Support/linux/Makefile
--- virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/Support/linux/Makefile	2009-10-07 11:27:14.000000000 +0200
+++ virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/Support/linux/Makefile	2009-10-07 13:43:30.705472033 +0200
@@ -44,9 +44,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
diff -urNad virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile
--- virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile	2009-10-07 11:27:14.000000000 +0200
+++ virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/VBoxNetAdp/linux/Makefile	2009-10-07 13:43:30.705472033 +0200
@@ -35,9 +35,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
diff -urNad virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile
--- virtualbox-ose-3.0.8-dfsg~/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile	2009-10-07 11:27:14.000000000 +0200
+++ virtualbox-ose-3.0.8-dfsg/src/VBox/HostDrivers/VBoxNetFlt/linux/Makefile	2009-10-07 13:43:30.705472033 +0200
@@ -35,9 +35,21 @@
   BUILD_TARGET_ARCH := amd64
  else
   ifeq ($(ARCH),i386)
-   BUILD_TARGET_ARCH := x86
+   ifeq ($(CONFIG_X86_32),y)
+     BUILD_TARGET_ARCH := x86
+   else
+     BUILD_TARGET_ARCH := amd64
+   endif
   else
-   BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   ifeq ($(ARCH),x86)
+     ifeq ($(CONFIG_X86_32),y)
+       BUILD_TARGET_ARCH := x86
+     else
+       BUILD_TARGET_ARCH := amd64
+     endif
+   else
+     BUILD_TARGET_ARCH := $(BUILD_TARGET_ARCH_DEF)
+   endif
   endif
  endif
 else
