Description: Adjusts failure message with Debian specific solution.
Author: Daniel Baumann <daniel@debian.org>

Index: virtualbox/src/VBox/VMM/VMMR3/VM.cpp
===================================================================
--- virtualbox.orig/src/VBox/VMM/VMMR3/VM.cpp
+++ virtualbox/src/VBox/VMM/VMMR3/VM.cpp
@@ -399,9 +399,11 @@
                     break;
                 case VERR_VERSION_MISMATCH:
                 case VERR_VM_DRIVER_VERSION_MISMATCH:
-                    pszError = N_("The VirtualBox support driver which is running is from a different "
-                                  "version of VirtualBox.  You can correct this by stopping all "
-                                  "running instances of VirtualBox and reinstalling the software.");
+                    pszError = N_("The version of the VirtualBox kernel modules and the version of "
+				  "VirtualBox application are not matching. You can correct this by "
+				  "either installing the correct virtualbox-modules package "
+				  "(if available) through apt-get or by building it manually with: "
+				  "module-assistant auto-install virtualbox");
                     break;
                 default:
                     pszError = N_("Unknown error initializing kernel driver");
