Description: Disable "Check for Updates" action.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/272212
Author: Daniel Hahler <ubuntu@thequod.de>, Felix Geyer <fgeyer@debian.org>, Gianfranco Costamagna <locutusofborg@debian.org>

Index: virtualbox/doc/manual/en_US/user_Introduction.xml
===================================================================
--- virtualbox.orig/doc/manual/en_US/user_Introduction.xml
+++ virtualbox/doc/manual/en_US/user_Introduction.xml
@@ -4405,14 +4405,14 @@
           <xref linkend="specialcharacters"/>.
         </para>
       </listitem>
-
+      <!--
       <listitem>
         <para>
           <emphasis role="bold">Update.</emphasis> Enables the user to
           specify various settings for Automatic Updates.
         </para>
       </listitem>
-
+      -->
       <listitem>
         <para>
           <emphasis role="bold">Language.</emphasis> Enables the user to
Index: virtualbox/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp
===================================================================
--- virtualbox.orig/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp
+++ virtualbox/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp
@@ -2349,7 +2349,7 @@
 bool UIExtraDataManager::applicationUpdateEnabled()
 {
     /* 'True' unless 'restriction' feature allowed: */
-    return !isFeatureAllowed(GUI_PreventApplicationUpdate);
+    return false;
 }
 
 QString UIExtraDataManager::applicationUpdateData()
Index: virtualbox/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
===================================================================
--- virtualbox.orig/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
+++ virtualbox/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
@@ -33,10 +33,10 @@
 #include "UIGlobalSettingsInput.h"
 #include "UIGlobalSettingsLanguage.h"
 #include "UIGlobalSettingsNetwork.h"
-#ifdef VBOX_GUI_WITH_NETWORK_MANAGER
-# include "UIGlobalSettingsProxy.h"
-# include "UIGlobalSettingsUpdate.h"
-#endif
+//#ifdef VBOX_GUI_WITH_NETWORK_MANAGER
+//# include "UIGlobalSettingsProxy.h"
+//# include "UIGlobalSettingsUpdate.h"
+//#endif
 
 /* GUI includes: Machine Settings: */
 #include "UIMachineSettingsAudio.h"
@@ -87,7 +87,7 @@
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     /* Update page: */
-    m_pSelector->setItemText(GlobalSettingsPageType_Update, tr("Update"));
+    //m_pSelector->setItemText(GlobalSettingsPageType_Update, tr("Update"));
 #endif
 
     /* Language page: */
@@ -104,7 +104,7 @@
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     /* Proxy page: */
-    m_pSelector->setItemText(GlobalSettingsPageType_Proxy, tr("Proxy"));
+    //m_pSelector->setItemText(GlobalSettingsPageType_Proxy, tr("Proxy"));
 #endif
 
     /* Polish the selector: */
@@ -205,13 +205,13 @@
                 }
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
                 /* Update page: */
-                case GlobalSettingsPageType_Update:
+                /*case GlobalSettingsPageType_Update:
                 {
                     pSettingsPage = new UIGlobalSettingsUpdate;
                     addItem(":/refresh_32px.png", ":/refresh_24px.png", ":/refresh_16px.png",
                             iPageIndex, "#update", pSettingsPage);
                     break;
-                }
+                }*/
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
                 /* Language page: */
                 case GlobalSettingsPageType_Language:
@@ -247,13 +247,13 @@
                 }
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
                 /* Proxy page: */
-                case GlobalSettingsPageType_Proxy:
+                /*case GlobalSettingsPageType_Proxy:
                 {
                     pSettingsPage = new UIGlobalSettingsProxy;
                     addItem(":/proxy_32px.png", ":/proxy_24px.png", ":/proxy_16px.png",
                             iPageIndex, "#proxy", pSettingsPage);
                     break;
-                }
+                }*/
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
                 default:
                     break;
