Description: Disable "Check for Updates" action.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/272212
Author: Daniel Hahler <ubuntu@thequod.de>, Felix Geyer <debfx-pkg@fobos.de>

diff -Nur virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
--- virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2010-04-27 22:24:11.000000000 +0200
+++ virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2010-05-03 22:05:38.752788747 +0200
@@ -4728,6 +4728,9 @@
  */
 void VBoxGlobal::showUpdateDialog (bool aForce)
 {
+    // Updates are handled through package managers.
+    return;
+
     /* Silently check in one day after current time-stamp */
     QTimer::singleShot (24 /* hours */   * 60   /* minutes */ *
                         60 /* seconds */ * 1000 /* milliseconds */,
diff -Nur virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp
--- virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp	2010-04-27 22:24:05.000000000 +0200
+++ virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp	2010-05-03 23:45:11.234038121 +0200
@@ -218,13 +218,6 @@
 #ifdef Q_WS_MAC
     if (m_fIsFirstTime)
 #endif /* Q_WS_MAC */
-        pMenu->addAction(pActionsPool->action(UIActionIndex_Simple_Update));
-
-#ifndef Q_WS_MAC
-    pMenu->addSeparator();
-#else /* Q_WS_MAC */
-    if (m_fIsFirstTime)
-#endif /* !Q_WS_MAC */
         pMenu->addAction(pActionsPool->action(UIActionIndex_Simple_About));
 
     /* Because this connections are done to VBoxGlobal, they are needed once only.
diff -Nur virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/settings/VBoxSettingsDialogSpecific.cpp virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/settings/VBoxSettingsDialogSpecific.cpp
--- virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/settings/VBoxSettingsDialogSpecific.cpp	2010-04-27 22:24:11.000000000 +0200
+++ virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/settings/VBoxSettingsDialogSpecific.cpp	2010-05-03 22:10:25.501539426 +0200
@@ -209,6 +209,8 @@
             /* Break to common result */
             break;
         }
+        case UpdateId:
+            return false;
         default:
             break;
     }
diff -Nur virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp
--- virtualbox-ose-3.1.51.orig/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp	2010-04-27 22:24:12.000000000 +0200
+++ virtualbox-ose-3.1.51/src/VBox/Frontends/VirtualBox/src/VBoxHelpActions.cpp	2010-05-03 22:04:06.154038535 +0200
@@ -88,13 +88,6 @@
         GetExtraData (VBoxDefs::GUI_RegistrationDlgWinID).isEmpty());
 #endif
 
-    aMenu->addAction (updateAction);
-    updateAction->setEnabled (vboxGlobal().virtualBox().
-        GetExtraData (VBoxDefs::GUI_UpdateDlgWinID).isEmpty());
-
-#ifndef Q_WS_MAC
-    aMenu->addSeparator();
-#endif /* Q_WS_MAC */
     aMenu->addAction (aboutAction);
 }
 
