Description: Disable "Check for Updates" action.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/272212
Author: Daniel Hahler <ubuntu@thequod.de>, Felix Geyer <fgeyer@debian.org>, Gianfranco Costamagna <locutusofborg@debian.org>

Index: virtualbox/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp
===================================================================
--- virtualbox.orig/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp
+++ virtualbox/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp
@@ -1038,7 +1038,7 @@
     /** Returns whether action is allowed. */
     virtual bool isAllowed() const /* override */
     {
-        return actionPool()->isAllowedInMenuApplication(UIExtraDataMetaDefs::MenuApplicationActionType_CheckForUpdates);
+        return false;
     }
 
     /** Returns shortcut extra-data ID. */
Index: virtualbox/doc/manual/en_US/user_Introduction.xml
===================================================================
--- virtualbox.orig/doc/manual/en_US/user_Introduction.xml
+++ virtualbox/doc/manual/en_US/user_Introduction.xml
@@ -3390,14 +3390,14 @@
           <xref linkend="specialcharacters"/>.
         </para>
       </listitem>
-
+      <!--
       <listitem>
         <para>
           <emphasis role="bold">Update.</emphasis> Enables the user to
           specify various settings for Automatic Updates.
         </para>
       </listitem>
-
+      -->
       <listitem>
         <para>
           <emphasis role="bold">Language.</emphasis> Enables the user to
