Description: Fix 3d guest acceleration with the multiarched mesa package.
Author: Felix Geyer <debfx-pkg@fobos.de>

diff --git a/src/VBox/Additions/common/crOpenGL/fakedri_drv.c b/src/VBox/Additions/common/crOpenGL/fakedri_drv.c
--- a/src/VBox/Additions/common/crOpenGL/fakedri_drv.c
+++ b/src/VBox/Additions/common/crOpenGL/fakedri_drv.c
@@ -37,10 +37,10 @@
 
 //@todo this could be different...
 #ifdef RT_ARCH_AMD64
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib64/dri:/usr/lib/dri"
+# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/x86_64-linux-gnu/dri:/usr/lib64/dri:/usr/lib/dri"
 # define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
 #else
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/dri"
+# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/i386-linux-gnu/dri:/usr/lib/dri"
 # define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
 #endif
 
