Description: Link against the static gsoap++ library.
Author: Felix Geyer <debfx-pkg@fobos.de>
Bug-Debian: http://bugs.debian.org/632368

diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -2021,16 +2021,22 @@ check_gsoap()
           fi
         fi
         if [ -f "$GSOAP_IMPORT/stlvector.h" ]; then
-          cnf_append "VBOX_GSOAP_INSTALLED"   "1"
-          cnf_append "VBOX_PATH_GSOAP"        "$GSOAP"
-          cnf_append "VBOX_PATH_GSOAP_IMPORT" "$GSOAP_IMPORT"
-          if [ -f "$GSOAP/share/gsoap/stdsoap2.cpp" ]; then
-            cnf_append "VBOX_GSOAP_CXX_SOURCES" "$GSOAP/share/gsoap/stdsoap2.cpp"
+          GSOAP_LIBDIR=`pkg-config --variable=libdir gsoap++`
+          if [ $? -eq 0 ] && [ -f "$GSOAP_LIBDIR/libgsoap++.a" ]; then
+            cnf_append "VBOX_GSOAP_INSTALLED"   "1"
+            cnf_append "VBOX_PATH_GSOAP"        "$GSOAP"
+            cnf_append "VBOX_PATH_GSOAP_IMPORT" "$GSOAP_IMPORT"
+            if [ -f "$GSOAP/share/gsoap/stdsoap2.cpp" ]; then
+              cnf_append "VBOX_GSOAP_CXX_SOURCES" "$GSOAP/share/gsoap/stdsoap2.cpp"
+            else
+              cnf_append "VBOX_GSOAP_CXX_SOURCES" ""
+            fi
+            cnf_append "VBOX_GSOAP_CXX_LIBS"    "$GSOAP_LIBDIR/libgsoap++.a"
+            log_success "found"
           else
-            cnf_append "VBOX_GSOAP_CXX_SOURCES" ""
+            log_failure "libgsoap++.a not found -- disabling webservice"
+            cnf_append "VBOX_WITH_WEBSERVICES" ""
           fi
-          cnf_append "VBOX_GSOAP_CXX_LIBS"    "libgsoap++"
-          log_success "found"
         else
           log_failure "stlvector.h not found -- disabling webservice"
           cnf_append "VBOX_WITH_WEBSERVICES" ""
