Description: Fix VBoxPci build failure.
Author: Felix Geyer <debfx-pkg@fobos.de>

--- VirtualBox-4.1.2_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk
+++ VirtualBox-4.1.2_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk
@@ -23,6 +23,7 @@ if1of ($(KBUILD_TARGET), linux)
  # The driver.
  # Note! For Linux, this is just a compile test. Don't use the binary!
  #
+ifdef VBOX_WITH_VBOXDRV
  SYSMODS += VBoxPci
  VBoxPci_TEMPLATE       = VBOXR0DRV
  VBoxPci_INST           = $(INST_VBOXPCI)$(if $(eq $(KBUILD_TARGET),darwin),Contents/MacOS/)
@@ -40,6 +41,7 @@ if1of ($(KBUILD_TARGET), linux)
  	VBoxPci.c
  VBoxPci_LIBS          += \
  	$(PATH_STAGE_LIB)/SUPR0IdcClient$(VBOX_SUFF_LIB)
+endif # VBOX_WITH_VBOXDRV
 
 
  #
