Description: Fix build errors with kernel 3.2.
Origin: upstream, https://www.virtualbox.org/ticket/9743

--- a/src/VBox/Additions/linux/sharedfolders/utils.c
+++ b/src/VBox/Additions/linux/sharedfolders/utils.c
@@ -110,7 +110,11 @@ void sf_init_inode(struct sf_glob_info *
         inode->i_fop   = &sf_dir_fops;
         /* XXX: this probably should be set to the number of entries
            in the directory plus two (. ..) */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 2, 0)
+        set_nlink(inode, 1);
+#else
         inode->i_nlink = 1;
+#endif
     }
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 0)
     else if (RTFS_IS_SYMLINK(attr->fMode))
@@ -119,7 +123,11 @@ void sf_init_inode(struct sf_glob_info *
         inode->i_mode &= ~sf_g->fmask;
         inode->i_mode |= S_IFLNK;
         inode->i_op    = &sf_lnk_iops;
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 2, 0)
+        set_nlink(inode, 1);
+# else
         inode->i_nlink = 1;
+# endif
     }
 #endif
     else
@@ -129,7 +137,11 @@ void sf_init_inode(struct sf_glob_info *
         inode->i_mode |= S_IFREG;
         inode->i_op    = &sf_reg_iops;
         inode->i_fop   = &sf_reg_fops;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 2, 0)
+        set_nlink(inode, 1);
+#else
         inode->i_nlink = 1;
+#endif
     }
 
     inode->i_uid = sf_g->uid;
--- a/src/VBox/HostDrivers/VBoxPci/linux/VBoxPci-linux.c
+++ b/src/VBox/HostDrivers/VBoxPci/linux/VBoxPci-linux.c
@@ -33,14 +33,22 @@
 #include "VBoxPciInternal.h"
 
 #ifdef VBOX_WITH_IOMMU
-#include <linux/dmar.h>
-#include <linux/intel-iommu.h>
-#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 1, 0)
-# include <asm/amd_iommu.h>
-#else
-# include <linux/amd-iommu.h>
-#endif
-#endif
+# include <linux/dmar.h>
+# include <linux/intel-iommu.h>
+# include <linux/pci.h>
+# if LINUX_VERSION_CODE < KERNEL_VERSION(3, 1, 0)
+#  include <asm/amd_iommu.h>
+# else
+#  include <linux/amd-iommu.h>
+# endif
+# if LINUX_VERSION_CODE < KERNEL_VERSION(3, 2, 0)
+#  define IOMMU_PRESENT()      iommu_found()
+#  define IOMMU_DOMAIN_ALLOC() iommu_domain_alloc()
+# else
+#  define IOMMU_PRESENT()      iommu_present(&pci_bus_type)
+#  define IOMMU_DOMAIN_ALLOC() iommu_domain_alloc(&pci_bus_type)
+# endif
+#endif /* VBOX_WITH_IOMMU */
 
 
 /*******************************************************************************
@@ -146,7 +154,7 @@ static int __init VBoxPciLinuxInit(void)
 #endif
 
 #ifdef VBOX_WITH_IOMMU
-    if (iommu_found())
+    if (IOMMU_PRESENT())
         printk(KERN_INFO "vboxpci: IOMMU found\n");
     else
         printk(KERN_INFO "vboxpci: IOMMU not found (not registered)\n");
@@ -984,9 +992,9 @@ int  vboxPciOsInitVm(PVBOXRAWPCIDRVVM pT
     printk(KERN_DEBUG "vboxPciOsInitVm: %p\n", pThis);
 #endif
 #ifdef VBOX_WITH_IOMMU
-    if (iommu_found())
+    if (IOMMU_PRESENT())
     {
-        pThis->pIommuDomain = iommu_domain_alloc();
+        pThis->pIommuDomain = IOMMU_DOMAIN_ALLOC();
         if (!pThis->pIommuDomain)
         {
             printk(KERN_DEBUG "cannot allocate IOMMU domain\n");
