Description: Fix FTBFS with gcc 4.7 on i386.
Origin: upstream, https://www.virtualbox.org/changeset/40420/vbox

--- a/src/libs/xpcom18a4/xpcom/io/nsFastLoadFile.h
+++ b/src/libs/xpcom18a4/xpcom/io/nsFastLoadFile.h
@@ -294,6 +294,6 @@
     // not call through our overridden Read method -- it calls directly into
     // the underlying input stream.
-    NS_IMETHODIMP ReadSegments(nsWriteSegmentFun aWriter, void* aClosure,
-                               PRUint32 aCount, PRUint32 *aResult);
+    NS_IMETHOD ReadSegments(nsWriteSegmentFun aWriter, void* aClosure,
+                            PRUint32 aCount, PRUint32 *aResult);
 
     nsresult ReadHeader(nsFastLoadHeader *aHeader);
--- a/src/VBox/Runtime/generic/critsect-generic.cpp
+++ b/src/VBox/Runtime/generic/critsect-generic.cpp
@@ -455,5 +455,5 @@
 
 
-RTDECL(int) RTCritSectEnterMultipleDebug(size_t cCritSects, PRTCRITSECT *papCritSects, RTUINTPTR uId, RT_SRC_POS_DECL)
+RTDECL(int) RTCritSectEnterMultipleDebug(size_t cCritSects, PRTCRITSECT *papCritSects, RTHCUINTPTR uId, RT_SRC_POS_DECL)
 {
     RTLOCKVALSRCPOS SrcPos = RTLOCKVALSRCPOS_INIT_DEBUG_API();
