Description: enable vboximg-mount build
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2020-07-15

--- virtualbox-6.1.12-dfsg.orig/Config.kmk
+++ virtualbox-6.1.12-dfsg/Config.kmk
@@ -1260,7 +1260,7 @@ ifdef VBOX_OSE
   VBOX_WITH_CLOUD_NET=
  endif
  # Disable building and packing vboximg-mount.
- VBOX_WITH_VBOXIMGMOUNT=
+ #VBOX_WITH_VBOXIMGMOUNT=
  VBOX_WITH_FUSE_PACKING=
  # branding
  VBOX_BRAND_LICENSE_HTML             := $(PATH_ROOT)/doc/License-gpl-2.0.html
