Index: src/VBox/Frontends/VirtualBox/Makefile.kmk
===================================================================
--- a/src/VBox/Frontends/VirtualBox/Makefile.kmk
+++ b/src/VBox/Frontends/VirtualBox/Makefile.kmk
@@ -1481,49 +1481,24 @@
 
 # Resource files with some OSE differences (VirtualBoxBrand.qrc is created further down).
-VirtualBox_SOURCES            += VirtualBox1.qrc
-VirtualBox1.qrc_RCCFLAGS       = -name BASIC1
-VirtualBox_SOURCES            += VirtualBox2.qrc
-VirtualBox2.qrc_RCCFLAGS       = -name BASIC2
-VirtualBox_SOURCES             += VirtualBox1_x2.qrc
+VBoxGlobal_SOURCES          += VirtualBox1.qrc
+VirtualBox1.qrc_RCCFLAGS     = -name BASIC1
+VBoxGlobal_SOURCES          += VirtualBox2.qrc
+VirtualBox2.qrc_RCCFLAGS     = -name BASIC2
+VBoxGlobal_SOURCES          += VirtualBox1_x2.qrc
 VirtualBox1_x2.qrc_RCCFLAGS  = -name BASIC1_X2
-VirtualBox_SOURCES             += VirtualBox2_x2.qrc
+VBoxGlobal_SOURCES          += VirtualBox2_x2.qrc
 VirtualBox2_x2.qrc_RCCFLAGS  = -name BASIC2_X2
-VirtualBox_SOURCES             += VirtualBox1_x3.qrc
+VBoxGlobal_SOURCES          += VirtualBox1_x3.qrc
 VirtualBox1_x3.qrc_RCCFLAGS  = -name BASIC1_X3
-VirtualBox_SOURCES             += VirtualBox2_x3.qrc
+VBoxGlobal_SOURCES          += VirtualBox2_x3.qrc
 VirtualBox2_x3.qrc_RCCFLAGS  = -name BASIC2_X3
-VirtualBox_SOURCES             += VirtualBox1_x4.qrc
+VBoxGlobal_SOURCES          += VirtualBox1_x4.qrc
 VirtualBox1_x4.qrc_RCCFLAGS  = -name BASIC1_X4
-VirtualBox_SOURCES             += VirtualBox2_x4.qrc
+VBoxGlobal_SOURCES          += VirtualBox2_x4.qrc
 VirtualBox2_x4.qrc_RCCFLAGS  = -name BASIC2_X4
-VirtualBox_SOURCES += $(VBOX_VIRTUALBOX_OUT_DIR)/VirtualBoxBrand.qrc
+VBoxGlobal_SOURCES += $(VBOX_VIRTUALBOX_OUT_DIR)/VirtualBoxBrand.qrc
 $(VBOX_VIRTUALBOX_OUT_DIR)/VirtualBoxBrand.qrc_RCCFLAGS = -name BRAND
 if defined(VBOX_WITH_VIDEOHWACCEL) || defined(VBOX_GUI_USE_QGL)
- VirtualBox_SOURCES += VirtualBoxShaders.qrc
- VirtualBoxShaders.qrc_RCCFLAGS = -name SHADERS
-endif
-
-
-# Resource files with some OSE differences (VirtualBoxBrand.qrc is created further down).
-VirtualBoxVM_SOURCES            += VirtualBox1.qrc
-VirtualBox1.qrc_RCCFLAGS       = -name BASIC1
-VirtualBoxVM_SOURCES            += VirtualBox2.qrc
-VirtualBox2.qrc_RCCFLAGS       = -name BASIC2
-VirtualBoxVM_SOURCES             += VirtualBox1_x2.qrc
-VirtualBox1_x2.qrc_RCCFLAGS  = -name BASIC1_X2
-VirtualBoxVM_SOURCES             += VirtualBox2_x2.qrc
-VirtualBox2_x2.qrc_RCCFLAGS  = -name BASIC2_X2
-VirtualBoxVM_SOURCES             += VirtualBox1_x3.qrc
-VirtualBox1_x3.qrc_RCCFLAGS  = -name BASIC1_X3
-VirtualBoxVM_SOURCES             += VirtualBox2_x3.qrc
-VirtualBox2_x3.qrc_RCCFLAGS  = -name BASIC2_X3
-VirtualBoxVM_SOURCES             += VirtualBox1_x4.qrc
-VirtualBox1_x4.qrc_RCCFLAGS  = -name BASIC1_X4
-VirtualBoxVM_SOURCES             += VirtualBox2_x4.qrc
-VirtualBox2_x4.qrc_RCCFLAGS  = -name BASIC2_X4
-VirtualBoxVM_SOURCES += $(VBOX_VIRTUALBOX_OUT_DIR)/VirtualBoxBrand.qrc
-$(VBOX_VIRTUALBOX_OUT_DIR)/VirtualBoxBrand.qrc_RCCFLAGS = -name BRAND
-if defined(VBOX_WITH_VIDEOHWACCEL) || defined(VBOX_GUI_USE_QGL)
- VirtualBoxVM_SOURCES += VirtualBoxShaders.qrc
+ VBoxGlobal_SOURCES += VirtualBoxShaders.qrc
  VirtualBoxShaders.qrc_RCCFLAGS = -name SHADERS
 endif
