Description: fix build with kernel 5.0
Origin: upstream
Index: virtualbox/src/VBox/Additions/linux/sharedfolders/vfsmod.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/sharedfolders/vfsmod.c
+++ virtualbox/src/VBox/Additions/linux/sharedfolders/vfsmod.c
@@ -40,7 +40,9 @@
 #include "revision-generated.h"
 #include "product-generated.h"
 #include "VBoxGuestR0LibInternal.h"
-#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
+# include <uapi/linux/mount.h> /* for MS_REMOUNT */
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0)
 # include <linux/mount.h>
 #endif
 #include <linux/seq_file.h>
