Description: Fix build of vboxvideo with kernel 5.11
Origin: upstream
Last-Update: 2021-04-08

--- virtualbox-6.1.18-dfsg.orig/src/VBox/Additions/linux/drm/vbox_drv.c
+++ virtualbox-6.1.18-dfsg/src/VBox/Additions/linux/drm/vbox_drv.c
@@ -362,7 +362,9 @@ static struct drm_driver driver = {
 
 #if RTLNX_VER_MAX(5,11,0)
 	.dev_priv_size = 0,
+# if RTLNX_VER_MIN(4,7,0)
 	.gem_free_object_unlocked = vbox_gem_free_object,
+# endif
 	.gem_prime_export = drm_gem_prime_export,
 	.gem_prime_pin = vbox_gem_prime_pin,
 	.gem_prime_unpin = vbox_gem_prime_unpin,
