Description: Check for system lzma library
Author: Gianfranco Costamagna <locutusofborg@debian.org>

--- virtualbox-7.0.8-dfsg.orig/configure
+++ virtualbox-7.0.8-dfsg/configure
@@ -110,6 +110,7 @@ WITH_DOCS=1
 WITH_LIBVPX=1
 WITH_LIBOGG=0
 WITH_LIBVORBIS=0
+WITH_LIBLZMA=1
 WITH_LIBTPMS=1
 BUILD_LIBXML2=
 BUILD_LIBCURL=
@@ -1805,6 +1810,44 @@ EOF
   fi
 }
 
+#
+# Check for liblzma
+#
+check_liblzma()
+{
+  if [ -z "$BUILD_LIBLZMA" ]; then
+    test_header liblzma
+    if which_wrapper pkg-config > /dev/null; then
+      libtpms_ver=`pkg-config liblzma --modversion 2>> $LOG`
+      if [ $? -eq 0 ]; then
+        FLGLZMA=`pkg-config liblzma --cflags`
+        INCLZMA=`strip_I "$FLGLZMA"`
+        LIBLZMA=`pkg-config liblzma --libs`
+      fi
+      cat > $ODIR.tmp_src.cc << EOF
+#include <cstdio>
+#include <lzma.h>
+extern "C" int main(void)
+{
+  lzma_stream strm = LZMA_STREAM_INIT;
+  uint32_t preset;
+  lzma_ret ret = lzma_easy_encoder(&strm, preset, LZMA_CHECK_CRC64);
+  printf("found, OK.\n");
+}
+EOF
+      [ -n "$INCLZMA" ] && I_INCLZMA=`prefix_I "$INCLZMA"`
+      if test_compile "$LIBLZMA $I_INCLZMA" liblzma liblzma nofatal; then
+        if test_execute; then
+          cnf_append "SDK_VBoxLibLzma_INCS" "$INCLZMA"
+          cnf_append "SDK_VBoxLibLzma_LIBS" "`strip_l "$LIBLZMA"`"
+        fi
+      else
+        echo "not found -- building liblzma from in-tree code."
+      fi
+    fi
+  fi
+}
+
 
 #
 # Check for libvorbis
@@ -2473,6 +2516,7 @@ EOF
 [ $WITH_LIBVORBIS -eq 0 ] && echo "  --enable-libvorbis       enable system libvorbis"
 [ $WITH_LIBOGG -eq 0 ]  && echo "  --enable-libogg          enable system libogg"
 [ $WITH_LIBTPMS -eq 1 ] && echo "  --disable-libtpms        don't use libtpms for TPM emulation"
+[ $WITH_LIBLZMA -eq 1 ] && echo "  --disable-liblzma        don't use liblzma"
 [ "$OS" = "linux" -o "$OS" = "freebsd" ] && echo "  --enable-vde             enable VDE networking"
 cat << EOF
   --disable-udptunnel      disable UDP tunnel networking
@@ -2481,6 +2525,7 @@ cat << EOF
   --build-libxml2          build libxml2 from sources
   --build-libssl           build openssl from sources
   --build-libtpms          build libtpms from sources
+  --build-liblzma          build liblzma from sources
 EOF
 [ $OSE -eq 0 ] && cat << EOF
   --build-libcurl          build libcurl from sources
@@ -2720,6 +2765,9 @@ for option in "$@"; do
     --disable-libtpms)
       WITH_LIBTPMS=0
       ;;
+    --disable-liblzma)
+      WITH_LIBLZMA=0
+      ;;
     --enable-libogg)
       WITH_LIBOGG=1
       ;;
@@ -2949,6 +2997,7 @@ if [ $ONLY_ADDITIONS -eq 0 ]; then
   [ $WITH_LIBOGG -eq 1 ] && check_libogg
   [ $WITH_LIBVORBIS -eq 1 ] && check_libvorbis
   [ $WITH_LIBTPMS -eq 1 ] && check_libtpms
+  [ $WITH_LIBLZMA -eq 1 ] && check_liblzma
   [ "$OS" != "darwin"  ] && check_png
   [ $OSE -eq 0 -a "$OS" = "linux" ] && check_pam
   if [ $WITH_SDL -eq 1 ]; then
