Index: virtualbox/src/VBox/Main/glue/glue-java.xsl
===================================================================
--- virtualbox.orig/src/VBox/Main/glue/glue-java.xsl
+++ virtualbox/src/VBox/Main/glue/glue-java.xsl
@@ -288,19 +288,15 @@
   <xsl:variable name="tagname" select="local-name()"/>
   <xsl:choose>
     <xsl:when test="$tagname = 'tt'">
-      <xsl:text>&lt;code</xsl:text>
+      <xsl:text>&lt;code&gt;</xsl:text>
     </xsl:when>
     <xsl:when test="$tagname = 'h3'">
-      <xsl:text>&lt;h2</xsl:text>
+      <xsl:text>&lt;h2&gt;</xsl:text>
     </xsl:when>
     <xsl:otherwise>
-      <xsl:value-of select="concat('&lt;', $tagname)"/>
+      <xsl:value-of select="concat('&lt;', $tagname, '&gt;')"/>
     </xsl:otherwise>
   </xsl:choose>
-  <xsl:if test="$tagname = 'table' and boolean($G_vboxFakeTableSummary)">
-    <xsl:text> summary=""</xsl:text>
-  </xsl:if>
-  <xsl:text>&gt;</xsl:text>
   <xsl:apply-templates/>
   <xsl:choose>
     <xsl:when test="$tagname = 'tt'">
@@ -491,28 +487,23 @@
 -->
 <xsl:template match="desc" mode="results">
   <xsl:if test="result">
-    <xsl:text>&#10;Expected result codes:&#10;</xsl:text>
-    <xsl:text>&lt;table</xsl:text>
-    <xsl:if test="boolean($G_vboxFakeTableSummary)">
-      <xsl:text> summary=""</xsl:text>
-    </xsl:if>
-    <xsl:text>&gt;&#10;</xsl:text>
+    <xsl:text>&#10;&lt;p&gt;&lt;/p&gt;&lt;dl&gt;&lt;dt&gt;&lt;b&gt;Expected result codes:&lt;/b&gt;&lt;/dt&gt;&#10;</xsl:text>
     <xsl:for-each select="result">
-      <xsl:text>&lt;tr&gt;</xsl:text>
+      <xsl:text>&lt;dd&gt;&lt;code&gt;</xsl:text>
       <xsl:choose>
         <xsl:when test="ancestor::library/result[@name=current()/@name]">
-          <xsl:value-of select="concat('&lt;td&gt;@link ::', @name, ' ', @name, '&lt;/td&gt;')"/>
+          <xsl:value-of select="concat('@link ::', @name, ' ', @name)"/>
         </xsl:when>
         <xsl:otherwise>
-          <xsl:value-of select="concat('&lt;td&gt;', @name, '&lt;/td&gt;')"/>
+          <xsl:value-of select="@name"/>
         </xsl:otherwise>
       </xsl:choose>
-      <xsl:text>&lt;td&gt;</xsl:text>
+      <xsl:text>&lt;/code&gt; - </xsl:text>
       <xsl:apply-templates select="text() | *[not(self::note or self::see or
                                                   self::result)]"/>
-      <xsl:text>&lt;/td&gt;&lt;/tr&gt;&#10;</xsl:text>
+      <xsl:text>&lt;/dd&gt;&#10;</xsl:text>
     </xsl:for-each>
-    <xsl:text>&lt;/table&gt;&#10;</xsl:text>
+    <xsl:text>&lt;/dl&gt;&#10;</xsl:text>
   </xsl:if>
 </xsl:template>
 
@@ -573,10 +564,10 @@
 <xsl:template match="desc" mode="method">
   <xsl:apply-templates select="." mode="begin"/>
   <xsl:apply-templates select="text() | *[not(self::note or self::see or self::result)]"/>
+  <xsl:apply-templates select="." mode="results"/>
   <xsl:for-each select="../param">
     <xsl:apply-templates select="desc"/>
   </xsl:for-each>
-  <xsl:apply-templates select="." mode="results"/>
   <xsl:apply-templates select="note"/>
   <xsl:apply-templates select="../param/desc/note"/>
   <xsl:apply-templates select="see"/>
Index: virtualbox/src/VBox/Main/webservice/Makefile.kmk
===================================================================
--- virtualbox.orig/src/VBox/Main/webservice/Makefile.kmk
+++ virtualbox/src/VBox/Main/webservice/Makefile.kmk
@@ -504,7 +504,6 @@
               --stringparam G_vboxApiSuffix $(VBOX_API_SUFFIX) \
               --stringparam G_vboxGlueStyle jaxws              \
               --stringparam G_vboxDirPrefix org/virtualbox$(VBOX_API_SUFFIX)/ \
-              --stringparam G_vboxFakeTableSummary $(if-expr $(VBOX_JAVA_VERSION) >= 10800 && $(VBOX_JAVA_VERSION) < 110000,1,"") \
               -o $(VBOX_JWS_GEN_RAWSRC) $(VBOX_GLUE_XSLT_DIR)/glue-java.xsl $<
 	$(QUIET)$(MKDIR) -p $(VBOX_JWS_GEN)/java/org/virtualbox$(VBOX_API_SUFFIX)
 	$(QUIET)$(VBOX_FILESPLIT) $(VBOX_JWS_GEN_RAWSRC) $(VBOX_JWS_GEN)/java
Index: virtualbox/src/VBox/Main/idl/VirtualBox.xidl
===================================================================
--- virtualbox.orig/src/VBox/Main/idl/VirtualBox.xidl
+++ virtualbox/src/VBox/Main/idl/VirtualBox.xidl
@@ -323,39 +323,35 @@
 
       The standard COM result codes that may originate from our methods include:
 
-      <table>
-      <tr><td>E_INVALIDARG</td>
-      <td>
-        Returned when the value of the method's argument is not within the range
-        of valid values. This should not be confused with situations when the
-        value is within the range but simply doesn't suit the current object
-        state and there is a possibility that it will be accepted later (in such
-        cases VirtualBox-specific codes are returned, for example,
-        <link to="VBOX_E_OBJECT_NOT_FOUND"/>).
-      </td>
-      </tr>
-      <tr><td>E_POINTER</td>
-      <td>
-        Returned if a memory pointer for the output argument is invalid (for
-        example, @c null). When pointers representing input arguments (such
-        as strings) are invalid, E_INVALIDARG is returned.
-      </td>
-      </tr>
-      <tr><td>E_ACCESSDENIED</td>
-      <td>
-        Returned when the called object is not ready. Since the lifetime of a
-        public COM object cannot be fully controlled by the implementation,
-        VirtualBox maintains the readiness state for all objects it creates and
-        returns this code in response to any method call on the object that was
-        deactivated by VirtualBox and is not functioning any more.
-      </td>
-      </tr>
-      <tr><td>E_OUTOFMEMORY</td>
-      <td>
-        Returned when a memory allocation operation fails.
-      </td>
-      </tr>
-      </table>
+      <dl>
+        <dt><tt>E_INVALIDARG</tt></dt>
+        <dd>
+          Returned when the value of the method's argument is not within the range
+          of valid values. This should not be confused with situations when the
+          value is within the range but simply doesn't suit the current object
+          state and there is a possibility that it will be accepted later (in such
+          cases VirtualBox-specific codes are returned, for example,
+          <link to="VBOX_E_OBJECT_NOT_FOUND"/>).
+        </dd>
+        <dt><tt>E_POINTER</tt></dt>
+        <dd>
+          Returned if a memory pointer for the output argument is invalid (for
+          example, @c null). When pointers representing input arguments (such
+          as strings) are invalid, E_INVALIDARG is returned.
+        </dd>
+        <dt><tt>E_ACCESSDENIED</tt></dt>
+        <dd>
+          Returned when the called object is not ready. Since the lifetime of a
+          public COM object cannot be fully controlled by the implementation,
+          VirtualBox maintains the readiness state for all objects it creates and
+          returns this code in response to any method call on the object that was
+          deactivated by VirtualBox and is not functioning any more.
+        </dd>
+        <dt><tt>E_OUTOFMEMORY</tt></dt>
+        <dd>
+          Returned when a memory allocation operation fails.
+        </dd>
+      </dl>
     </desc>
   </descGroup>
 
@@ -14942,30 +14938,18 @@
 
           The following flags are available:
 
-          <table>
-            <tr>
-              <td>CopyIntoExisting</td>
-              <td>Allow copying into an existing destination directory.</td>
-            </tr>
-          </table>
-
-          <table>
-            <tr>
-              <td>NoReplace</td>
-              <td>Do not replace any existing destination files on
-                the destination.</td>
-            </tr>
-            <tr>
-              <td>FollowLinks</td>
-              <td>Follows (and handles) (symbolic) links.</td>
-            </tr>
-            <tr>
-              <td>Update</td>
-              <td>Only copy when the source file is newer than the destination
-                file or when the destination file is missing.</td>
-            </tr>
-          </table>
-
+          <dl>
+            <dt><tt>CopyIntoExisting</tt></dt>
+            <dd>Allow copying into an existing destination directory.</dd>
+            <dt><tt>NoReplace</tt></dt>
+            <dd>Do not replace any existing destination files on
+              the destination.</dd>
+            <dt><tt>FollowLinks</tt></dt>
+            <dd>Follows (and handles) (symbolic) links.</dd>
+            <dt><tt>Update</tt></dt>
+            <dd>Only copy when the source file is newer than the destination
+              file or when the destination file is missing.</dd>
+          </dl>
         </desc>
       </param>
       <param name="destination" type="wstring" dir="in">
@@ -14997,29 +14981,18 @@
 
           The following flags are available:
 
-          <table>
-            <tr>
-              <td>CopyIntoExisting</td>
-              <td>Allow copying into an existing destination directory.</td>
-            </tr>
-          </table>
-
-          <table>
-            <tr>
-              <td>NoReplace</td>
-              <td>Do not replace any existing destination files on
-                the destination.</td>
-            </tr>
-            <tr>
-              <td>FollowLinks</td>
-              <td>Follows (and handles) (symbolic) links.</td>
-            </tr>
-            <tr>
-              <td>Update</td>
-              <td>Only copy when the source file is newer than the destination
-                file or when the destination file is missing.</td>
-            </tr>
-          </table>
+          <dl>
+            <dt><tt>CopyIntoExisting</tt></dt>
+            <dd>Allow copying into an existing destination directory.</dd>
+            <dt><tt>NoReplace</tt></dt>
+            <dd>Do not replace any existing destination files on
+              the destination.</dd>
+            <dt><tt>FollowLinks</tt></dt>
+            <dd>Follows (and handles) (symbolic) links.</dd>
+            <dt><tt>Update</tt></dt>
+            <dd>Only copy when the source file is newer than the destination
+              file or when the destination file is missing.</dd>
+          </dl>
 
         </desc>
       </param>
@@ -20356,12 +20329,12 @@
         <desc>
           The current state of mouse buttons. Every bit represents
           a mouse button as follows:
-          <table>
-            <tr><td>Bit 0 (<tt>0x01</tt>)</td><td>left mouse button</td></tr>
-            <tr><td>Bit 1 (<tt>0x02</tt>)</td><td>right mouse button</td></tr>
-            <tr><td>Bit 2 (<tt>0x04</tt>)</td><td>middle mouse button</td></tr>
-          </table>
-          A value of <tt>1</tt> means the corresponding button is pressed.
+          <dl>
+            <dt>Bit 0 (<tt>0x01</tt>)</dt><dd>left mouse button</dd>
+            <dt>Bit 1 (<tt>0x02</tt>)</dt><dd>right mouse button</dd>
+            <dt>Bit 2 (<tt>0x04</tt>)</dt><dd>middle mouse button</dd>
+          </dl>
+          A value of @c 1 means the corresponding button is pressed.
           otherwise it is released.
         </desc>
       </param>
@@ -20421,11 +20394,11 @@
         <desc>
           The current state of mouse buttons. Every bit represents
           a mouse button as follows:
-          <table>
-            <tr><td>Bit 0 (<tt>0x01</tt>)</td><td>left mouse button</td></tr>
-            <tr><td>Bit 1 (<tt>0x02</tt>)</td><td>right mouse button</td></tr>
-            <tr><td>Bit 2 (<tt>0x04</tt>)</td><td>middle mouse button</td></tr>
-          </table>
+          <dl>
+            <dt>Bit 0 (<tt>0x01</tt>)</dt><dd>left mouse button</dd>
+            <dt>Bit 1 (<tt>0x02</tt>)</dt><dd>right mouse button</dd>
+            <dt>Bit 2 (<tt>0x04</tt>)</dt><dd>middle mouse button</dd>
+          </dl>
           A value of @c 1 means the corresponding button is pressed.
           otherwise it is released.
         </desc>
