Bug-Debian: https://bugs.debian.org/934483
Last-Update: 2019-08-21

--- virtualbox-6.0.10-dfsg.orig/src/VBox/Additions/linux/drm/vbox_ttm.c
+++ virtualbox-6.0.10-dfsg/src/VBox/Additions/linux/drm/vbox_ttm.c
@@ -33,7 +33,7 @@
  *          Michael Thayer <michael.thayer@oracle.com>
  */
 #include "vbox_drv.h"
-#include <ttm/ttm_page_alloc.h>
+#include <drm/ttm/ttm_page_alloc.h>
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 18, 0) && !defined(RHEL_72)
 #define PLACEMENT_FLAGS(placement) (placement)
