--- virtualbox-5.0.20/src/VBox/HostDrivers/VBoxNetAdp/linux/VBoxNetAdp-linux.c.orig	2016-04-28 15:14:24.000000000 +0200
+++ virtualbox-5.0.20/src/VBox/HostDrivers/VBoxNetAdp/linux/VBoxNetAdp-linux.c		2016-05-30 11:07:38.806490593 +0200
@@ -153,7 +153,11 @@
     pPriv->Stats.tx_packets++;
     pPriv->Stats.tx_bytes += pSkb->len;
     /* Update transmission time stamp. */
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0)
+    netif_trans_update(pNetDev);
+# else
     pNetDev->trans_start = jiffies;
+# endif
     /* Nothing else to do, just free the sk_buff. */
     dev_kfree_skb(pSkb);
     return 0;
