Description: cherry-pick upstream changeset 57658
Author: Frank Mehnert <frank.mehnert@oracle.com>

Index: src/VBox/ExtPacks/VNC/VBoxVNC.cpp
===================================================================
--- a/src/VBox/ExtPacks/VNC/VBoxVNC.cpp
+++ b/src/VBox/ExtPacks/VNC/VBoxVNC.cpp
@@ -104,6 +104,6 @@
     unsigned char *mFrameBuffer;
     uint32_t uClients;
-    static DECLCALLBACK(enum rfbNewClientAction) rfbNewClientEvent(rfbClientPtr cl);
-    static DECLCALLBACK(void) vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
+    static enum rfbNewClientAction rfbNewClientEvent(rfbClientPtr cl);
+    static void vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
     static void vncKeyboardEvent(rfbBool down, rfbKeySym keySym, rfbClientPtr cl);
     static void clientGoneHook(rfbClientPtr cl);
