Description: Fix FTBFS with gcc-4.9
Author: Mateusz Łukasik <mati75@linuxmint.pl>
Bug-Debian: http://bugs.debian.org/754220

--- a/configure
+++ b/configure
@@ -415,9 +415,9 @@ check_gcc()
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 8 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
              -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<8"
+        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<9"
         fail really
       else
         log_success "found version $cc_ver"
