Author: Gianfranco Costamagna <locutusofborg@debian.org>

--- virtualbox-5.1.0~BETA1-dfsg.orig/configure
+++ virtualbox-5.1.0~BETA1-dfsg/configure
@@ -438,7 +438,7 @@ check_gcc()
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
-             -o \( $cc_maj -eq 5 -a $cc_min -gt 3 \) \
+             -o \( $cc_maj -eq 5 -a $cc_min -gt 4 \) \
              -o \( $cc_maj -eq 6 -a $cc_min -gt 1 \) \
              -o $cc_maj -gt 6 ]; then
         log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10, gcc 5.x with x<10 or gcc 6.x with x<2"
