Index: virtualbox/Config.kmk
===================================================================
--- virtualbox.orig/Config.kmk
+++ virtualbox/Config.kmk
@@ -7284,8 +7284,8 @@
 if defined(VBOX_JAVA_VERSION) && $(VBOX_JAVA_VERSION) >= 10900
  VBOX_JAVA_WS_OPTS ?= --add-modules java.xml.ws
 endif
-VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 1.5 -target 1.5 -Xlint:unchecked
-VBOX_JAVADOC_OPTS = -encoding UTF-8
+VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 1.8 -target 1.8 -Xlint:unchecked
+VBOX_JAVADOC_OPTS = -encoding UTF-8
 
 
 #
