Index: virtualbox/Config.kmk
===================================================================
--- virtualbox.orig/Config.kmk
+++ virtualbox/Config.kmk
@@ -7262,8 +7262,8 @@
 else
  VBOX_SEP = $(COLON)
 endif
-VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 1.5 -target 1.5 -Xlint:unchecked
-VBOX_JAVADOC_OPTS = -encoding UTF-8
+VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 1.9 -target 1.9 -Xlint:unchecked --add-modules java.xml.ws
+VBOX_JAVADOC_OPTS = -encoding UTF-8 --add-modules java.xml.ws
 
 
 #
