Index: virtualbox/Config.kmk
===================================================================
--- virtualbox.orig/Config.kmk
+++ virtualbox/Config.kmk
@@ -7745,14 +7745,14 @@
 
 # Java compiler options, needs some version checking.
 if defined(VBOX_JAVA_VERSION) && $(VBOX_JAVA_VERSION) >= 10900
- VBOX_JAVA_WS_OPTS ?= --add-modules java.xml.ws
+ VBOX_JAVA_WS_OPTS ?=
 endif
 if defined(VBOX_JAVA_VERSION) && $(VBOX_JAVA_VERSION) >= 100000
- VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 6 -target 6 -Xlint:unchecked
+ VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 9 -target 9 -Xlint:unchecked
 else
  VBOX_JAVAC_OPTS   = -encoding UTF-8 -source 1.5 -target 1.5 -Xlint:unchecked
 endif
-VBOX_JAVADOC_OPTS = -encoding UTF-8
+VBOX_JAVADOC_OPTS = -encoding UTF-8 -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar
 
 
 #
Index: virtualbox/src/VBox/Main/Makefile.kmk
===================================================================
--- virtualbox.orig/src/VBox/Main/Makefile.kmk
+++ virtualbox/src/VBox/Main/Makefile.kmk
@@ -1529,7 +1529,7 @@
 	$(QUIET)$(MKDIR) -p $(VBOX_JMSCOM_JDEST)
 	$(call MSG_TOOL,javac,$(notdir $@),...,)
 	$(QUIET)$(VBOX_JAVAC) $(VBOX_JAVAC_OPTS) @$(VBOX_JMSCOM_GEN)/jmscomglue.list \
-		-d $(VBOX_JMSCOM_JDEST) -classpath "$(VBOX_JMSCOM_JDEST)$(VBOX_SEP)$(VBOX_JACOB_DIR)/jacob.jar"
+		-d $(VBOX_JMSCOM_JDEST) -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar:"$(VBOX_JMSCOM_JDEST)$(VBOX_SEP)$(VBOX_JACOB_DIR)/jacob.jar"
 	$(call MSG_LINK,$(notdir $@),$@)
 	$(QUIET)$(VBOX_JAR) cf $@ -C $(VBOX_JMSCOM_JDEST) .
 
Index: virtualbox/src/VBox/Main/webservice/Makefile.kmk
===================================================================
--- virtualbox.orig/src/VBox/Main/webservice/Makefile.kmk
+++ virtualbox/src/VBox/Main/webservice/Makefile.kmk
@@ -480,7 +480,7 @@
 	$(call MSG_L1,Compiling bridge code)
 	$(VBOX_JAVAC) $(VBOX_JAVAC_OPTS) $(VBOX_JAVA_WS_OPTS) \
                 @$(VBOX_JWS_GEN)/jwsglue.list \
-		-d $(VBOX_JWS_JDEST) -classpath $(VBOX_JWS_JDEST)
+		-d $(VBOX_JWS_JDEST) -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar:$(VBOX_JWS_JDEST)
 	$(QUIET)$(SED) -e "s/vboxweb.wsdl/vboxweb$(VBOX_API_SUFFIX).wsdl/" <  $(VBOXWEBSERVICE_WSDL) > $(VBOX_JWS_JDEST)/vboxwebService$(VBOX_API_SUFFIX).wsdl
 	$(QUIET)$(CP) -f $(VBOXWEB_WSDL) $(VBOX_JWS_JDEST)/vboxweb$(VBOX_API_SUFFIX).wsdl
 	$(call MSG_LINK,$(notdir $@),$@)
Index: virtualbox/src/libs/xpcom18a4/java/Makefile.kmk
===================================================================
--- virtualbox.orig/src/libs/xpcom18a4/java/Makefile.kmk
+++ virtualbox/src/libs/xpcom18a4/java/Makefile.kmk
@@ -114,7 +114,7 @@
 #
 $(VBOX_JXPCOM_GENH): $$(VBOX_JXPCOM_JAR)
 	$(call MSG_L1,Generating $@ from $<)
-	$(QUIET)$(VBOX_JAVAH) -classpath $(VBOX_JXPCOM_JDEST) -d $(VBOX_JXPCOM_JDEST) \
+	$(QUIET)$(VBOX_JAVAH) -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar:$(VBOX_JXPCOM_JDEST) -d $(VBOX_JXPCOM_JDEST) \
 		org.mozilla.xpcom.internal.XPCOMImpl \
 		org.mozilla.xpcom.internal.GREImpl \
 		org.mozilla.xpcom.internal.MozillaImpl \
@@ -210,13 +210,13 @@
 	$(QUIET)$(RM) -Rf $(VBOX_JXPCOM_JDEST)
 	$(QUIET)$(MKDIR) -p $(VBOX_JXPCOM_JDEST)
 	$(QUIET)$(VBOX_JAVAC) $(VBOX_JAVAC_OPTS) @$(VBOX_JXPCOM_GEN)/jxpcomgen.list \
-		-d $(VBOX_JXPCOM_JDEST) -classpath $(VBOX_JXPCOM_JDEST)
+		-d $(VBOX_JXPCOM_JDEST) -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar:$(VBOX_JXPCOM_JDEST)
 	$(call MSG_TOOL,javac,$(notdir $@),...,)
 	$(QUIET)$(VBOX_JAVAC) $(VBOX_JAVAC_OPTS) \
 		$(VBOX_JXPCOM_JAR_SRC) \
 		$(VBOX_JXPCOM_NSERROR) \
                 @$(VBOX_JXPCOM_GEN)/jxpcomglue.list \
-		-d $(VBOX_JXPCOM_JDEST) -classpath $(VBOX_JXPCOM_JDEST)
+		-d $(VBOX_JXPCOM_JDEST) -classpath /usr/share/java/jws-api.jar:/usr/share/java/jaxb-api.jar:/usr/share/java/jaxws-api.jar:$(VBOX_JXPCOM_JDEST)
 	$(call MSG_LINK,$(notdir $@),$@)
 	$(QUIET)$(VBOX_JAR) cf $@ -C $(VBOX_JXPCOM_JDEST) .
 
