Description: Cherry-pick upstream changes for LaTeX build fixes.

Origin: https://www.virtualbox.org/changeset?reponame=vbox&new=96615%40trunk%2Fdoc%2Fmanual%2Fdocbook2latex.xsl&old=96407%40trunk%2Fdoc%2Fmanual%2Fdocbook2latex.xsl&format=diff

Index: doc/manual/docbook2latex.xsl
===================================================================
--- a/doc/manual/docbook2latex.xsl	(revision 96407)
+++ b/doc/manual/docbook2latex.xsl	(revision 96615)
@@ -94,6 +94,4 @@
 \usepackage{geometry}
 \geometry{top=3cm,bottom=4cm}
-\usepackage{ucs}
-\usepackage[utf8x]{inputenc}
 \usepackage[T1]{fontenc}
 \usepackage{tabulary}
@@ -1246,5 +1244,29 @@
           </xsl:call-template>
         </xsl:variable>
-        <xsl:value-of select="$subst13" />
+        <xsl:variable name="subst14">
+          <xsl:call-template name="str:subst">
+            <xsl:with-param name="text" select="$subst13" />
+            <xsl:with-param name="replace" select="'→'" />
+            <xsl:with-param name="with" select="'\ensuremath{\rightarrow}'" />
+            <xsl:with-param name="disable-output-escaping" select="no" />
+          </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="subst15">
+          <xsl:call-template name="str:subst">
+            <xsl:with-param name="text" select="$subst14" />
+            <xsl:with-param name="replace" select="'←'" />
+            <xsl:with-param name="with" select="'\ensuremath{\leftarrow}'" />
+            <xsl:with-param name="disable-output-escaping" select="no" />
+          </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="subst16">
+          <xsl:call-template name="str:subst">
+            <xsl:with-param name="text" select="$subst15" />
+            <xsl:with-param name="replace" select="'↔'" />
+            <xsl:with-param name="with" select="'\ensuremath{\leftrightarrow}'" />
+            <xsl:with-param name="disable-output-escaping" select="no" />
+          </xsl:call-template>
+        </xsl:variable>
+        <xsl:value-of select="$subst16" />
       </xsl:otherwise>
     </xsl:choose>
