Description: Use the packaged liblzf library
Author: Gianfranco Costamagna <locutusofborg@debian.org>

--- a/src/libs/Makefile.kmk
+++ b/src/libs/Makefile.kmk
@@ -24,10 +24,6 @@ include $(KBUILD_PATH)/subheader.kmk
 #       include path specific to the lib has been changed.
 #
 
-# Compression libs used by IPRT and <what was it again?>.
-if "$(SDK_VBOX_LZF_INCS)" == "$(PATH_ROOT)/src/libs/liblzf-3.4"
- include $(PATH_SUB_CURRENT)/liblzf-3.4/Makefile.kmk
-endif
 if  !defined(VBOX_ONLY_SDK) \
  && "$(SDK_VBOX_ZLIB_INCS)" == "$(PATH_ROOT)/src/libs/zlib-1.2.11"
  include $(PATH_SUB_CURRENT)/zlib-1.2.11/Makefile.kmk
--- a/src/VBox/Runtime/Makefile.kmk
+++ b/src/VBox/Runtime/Makefile.kmk
@@ -298,7 +298,7 @@ endif
 RuntimeR3_INCS          = \
 	include \
 	$(SDK_VBOX_ZLIB_INCS) \
-	$(SDK_VBOX_LZF_INCS)
+	/usr/include/liblzf
 
 # RuntimeR3_INCS.solaris  = \
 #    /usr/include
@@ -2205,7 +2205,7 @@ VBoxRT_INCS                   := $(Runti
 VBoxRT_INCS.$(KBUILD_TARGET)  := $(RuntimeR3_INCS.$(KBUILD_TARGET))
 VBoxRT_INCS.$(KBUILD_TARGET).$(KBUILD_TARGET_ARCH) := $(RuntimeR3_INCS.$(KBUILD_TARGET).$(KBUILD_TARGET_ARCH))
 VBoxRT_LIBS                    = \
-	$(SDK_VBOX_LZF_LIBS)
+	lzf
 if1of ($(KBUILD_TARGET_ARCH), amd64 x86)
 VBoxRT_LIBS                   += \
 	$(PATH_STAGE_LIB)/SUPR3$(VBOX_SUFF_LIB)
@@ -2352,7 +2352,7 @@ endif
 VBoxRT-x86_SDKS  += VBOX_OPENSSL-x86
 
 VBoxRT-x86_LIBS = \
-	$(SDK_VBOX_LZF_LIBS-x86)
+	lzf
 if1of ($(KBUILD_TARGET_ARCH), amd64 x86)
 VBoxRT-x86_LIBS += \
 	$(PATH_STAGE_LIB)/SUPR3-x86$(VBOX_SUFF_LIB)
--- a/src/VBox/Storage/testcase/Makefile.kmk
+++ b/src/VBox/Storage/testcase/Makefile.kmk
@@ -135,11 +135,11 @@ if defined(VBOX_WITH_TESTCASES) || defin
 	$(VBOX_LIB_RUNTIME_STATIC)
  if1of ($(KBUILD_TARGET),os2 win)
   vbox-img_LIBS += \
-	$(SDK_VBOX_LZF_STATIC_LIBS) \
+	lzf \
 	$(SDK_VBOX_ZLIB_STATIC_LIBS)
  else
   vbox-img_LIBS += \
-	$(SDK_VBOX_LZF_LIBS) \
+	lzf \
 	$(SDK_VBOX_ZLIB_LIBS)
  endif
  ifeq ($(KBUILD_TARGET),linux)
@@ -181,11 +181,11 @@ if    defined(VBOX_WITH_TESTCASES) && de
 	$(PATH_STAGE_LIB)/SUPR3$(VBOX_SUFF_LIB)
  if1of ($(KBUILD_TARGET),os2 win)
   vdkeystoremgr_LIBS += \
-	$(SDK_VBOX_LZF_STATIC_LIBS) \
+	lzf \
 	$(SDK_VBOX_ZLIB_STATIC_LIBS)
  else
   vdkeystoremgr_LIBS += \
-	$(SDK_VBOX_LZF_LIBS) \
+	lzf \
 	$(SDK_VBOX_ZLIB_LIBS)
  endif
  ifeq ($(KBUILD_TARGET),linux)
--- a/Config.kmk
+++ b/Config.kmk
@@ -5458,7 +5458,7 @@ if1of ($(KBUILD_TARGET),os2 win)
  	$(SDK_VBOX_ZLIB_STATIC_LIBS)
 else
  TEMPLATE_VBoxR3Static_LIBS        += \
- 	$(PATH_STAGE_LIB)/VBox-liblzf$(VBOX_SUFF_LIB) \
+ 	lzf \
  	$(SDK_VBOX_ZLIB_LIBS)
  if1of ($(KBUILD_TARGET),darwin freebsd)
   TEMPLATE_VBoxR3Static_LIBS       += iconv
