Index: virtualbox-5.1.14-dfsg/src/VBox/Additions/linux/sharedfolders/lnkops.c
===================================================================
--- virtualbox-5.1.14-dfsg.orig/src/VBox/Additions/linux/sharedfolders/lnkops.c
+++ virtualbox-5.1.14-dfsg/src/VBox/Additions/linux/sharedfolders/lnkops.c
@@ -90,7 +90,9 @@ static const char *sf_get_link(struct de
 
 struct inode_operations sf_lnk_iops =
 {
+# if LINUX_VERSION_CODE < KERNEL_VERSION(4, 10, 0)
     .readlink       = generic_readlink,
+# endif
 # if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 5, 0)
     .get_link       = sf_get_link
 # elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 2, 0)
