Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2018-07-19

--- virtualbox-5.2.16-dfsg.orig/configure
+++ virtualbox-5.2.16-dfsg/configure
@@ -434,8 +434,8 @@ check_gcc()
       elif [ $cc_maj -lt 4 \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 4 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 2 -a "$OS"  = "darwin" \) \
-             -o $cc_maj -gt 7 ]; then
-        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x...7.x"
+             -o $cc_maj -gt 8 ]; then
+        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 4.x...8.x"
         fail really
       else
         log_success "found version $cc_ver"
