Description: don't add VBoxREM symlink
Forwarded: irc
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Bug-Debian: https://bugs.debian.org/969793
Last-Update: 2020-09-08

--- virtualbox-6.1.14-dfsg.orig/src/VBox/Installer/linux/deffiles
+++ virtualbox-6.1.14-dfsg/src/VBox/Installer/linux/deffiles
@@ -32,7 +32,6 @@ DEFAULT_FILE_NAMES=" \
     VBoxManage \
     VBoxNetDHCP \
     VBoxNetDHCP.so \
-    VBoxREM.so \
     VBoxRT.so \
     VBoxSDL \
     VBoxSDL.so \
@@ -82,7 +81,6 @@ DEFAULT_FILE_NAMES=" \
     components/xpti.dat \
     components/VBoxC.so \
     components/VBoxDDU.so \
-    components/VBoxREM.so \
     components/VBoxRT.so \
     components/VBoxVMM.so \
     components/VBoxXPCOM.so \
--- virtualbox-6.1.14-dfsg.orig/src/VBox/Installer/linux/uninstall.sh
+++ virtualbox-6.1.14-dfsg/src/VBox/Installer/linux/uninstall.sh
@@ -96,7 +96,6 @@ rm -f \
   /usr/bin/vboxdtrace \
   /usr/bin/vboxbugreport \
   $PREV_INSTALLATION/components/VBoxVMM.so \
-  $PREV_INSTALLATION/components/VBoxREM.so \
   $PREV_INSTALLATION/components/VBoxRT.so \
   $PREV_INSTALLATION/components/VBoxDDU.so \
   $PREV_INSTALLATION/components/VBoxXPCOM.so \
--- virtualbox-6.1.14-dfsg.orig/src/VBox/Main/Makefile.kmk
+++ virtualbox-6.1.14-dfsg/src/VBox/Main/Makefile.kmk
@@ -1294,7 +1294,6 @@ if defined(VBOX_WITH_HARDENING) && "$(KB
  VBoxMain-hardening-inst_INST = $(INST_BIN)components/
  VBoxMain-hardening-inst_SYMLINKS = \
 	VBoxDDU.so=>../VBoxDDU.so \
-	VBoxREM.so=>../VBoxREM.so \
 	VBoxRT.so=>../VBoxRT.so \
 	VBoxVMM.so=>../VBoxVMM.so \
 	VBoxXPCOM.so=>../VBoxXPCOM.so
