Index: virtualbox/src/VBox/Main/webservice/Makefile.kmk
===================================================================
--- virtualbox.orig/src/VBox/Main/webservice/Makefile.kmk	2013-11-22 14:26:08.851312152 -0500
+++ virtualbox/src/VBox/Main/webservice/Makefile.kmk	2013-11-22 14:30:33.635324450 -0500
@@ -694,7 +694,7 @@
 ifdef VBOX_GSOAP_VERBOSE
 	$(MV) -f $(wildcard $(VBOXWEB_OUT_DIR)/*.req.xml $(VBOXWEB_OUT_DIR)/*.res.xml) $(PATH_TARGET_SOAPDEMOXML)
 endif
-	$(MV) -f $(wildcard $(VBOXWEB_OUT_DIR)/soapvbox*.h) $(PATH_TARGET_SOAPDEMOHEADERS)
+	#$(MV) -f $(wildcard $(VBOXWEB_OUT_DIR)/soapvbox*.h) $(PATH_TARGET_SOAPDEMOHEADERS)
 	$(MV) -f $(VBOXWEB_OUT_DIR)/vboxBinding.nsmap $(PATH_TARGET_SOAPDEMONSMAPS)
 	$(APPEND) $@ done
 
