Index: virtualbox/src/libs/xpcom18a4/python/vboxxpcom.py
===================================================================
--- virtualbox.orig/src/libs/xpcom18a4/python/vboxxpcom.py
+++ virtualbox/src/libs/xpcom18a4/python/vboxxpcom.py
@@ -39,7 +39,7 @@
 if sys.hexversion >= 0x030200f0 and sys.abiflags:
     _asNew = []
     for sCandidate in _asVBoxPythons:
-        if isdigit(sCandidate[-1:]):
+        if sCandidate[-1:].isdigit():
             _asNew.append(sCandidate + sys.abiflags)
         else:
             _asNew.append(sCandidate)
