Origin: https://www.virtualbox.org/ticket/17711

Index: src/VBox/Devices/VMMDev/VMMDevHGCM.cpp
===================================================================
--- a/src/VBox/Devices/VMMDev/VMMDevHGCM.cpp
+++ b/src/VBox/Devices/VMMDev/VMMDevHGCM.cpp
@@ -1822,4 +1822,5 @@
     /* Get local copy of the list of loaded commands. */
     RTLISTANCHOR listLoadedCommands;
+    RTListInit(&listLoadedCommands); /* RTListMove does not touch the destination is the source list is empty. */
     RTListMove(&listLoadedCommands, &pThis->listHGCMCmd);
