Index: VirtualBox-6.1.6/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp
===================================================================
--- virtualbox.orig/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp
+++ virtualbox/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp
@@ -293,7 +293,7 @@
     }
 }
 
-int x11MonitorThreadFunction(RTTHREAD hThreadSelf, void *pvUser)
+DECLCALLBACK(int) x11MonitorThreadFunction(RTTHREAD hThreadSelf, void *pvUser)
 {
     (void)hThreadSelf;
     (void*)pvUser;
